src/data/list/min_max.lean

```diff
@@ -315,6 +315,20 @@ begin
   { simpa [h y (mem_cons_self _ _)] using IH (λ x hx, h x $ mem_cons_of_mem _ hx) }
 end
 
+lemma le_max_of_exists_le (l : list α) (a : α) (h1 : l ≠ []) (h2 : ∃ x ∈ l, a ≤ x) :
```

Comment:
You don't need `h1` here. I'd suggest this spelling
```suggestion
lemma le_max_of_mem_of_le (l : list α) (a : α) {x : α} (h2 : x ∈ l) (h_le : a ≤ x) :
```
let see what @YaelDillies  thinks

Reply:
I would personally go for
```suggestion
lemma le_max_of_mem_of_le {l : list α} {a x : α} (hx : x ∈ l) (h : a ≤ x) :
```

Reply:
I've removed the `h1` hypothesis. I've kept `a` and `l` explicit for consistency with the previous lemma in the file (although my preference would also be to make them implicit). Should I just change this anyway?

Reply:
Yes please! I don't quite like the explicitness in the previous lemma but it's slightly more justified by the fact that `l` and `a` ionly appear within a binder.

Reply:
Done!